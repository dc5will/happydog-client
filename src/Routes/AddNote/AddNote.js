import React, { Component } from 'react';
import config from '../../config'

export default class LandingPage extends Component {
    constructor(props) {
        super(props);
        this.state = {  
            note_name: '',
            nameValid: false,
            nameValidationMessage: '',
            contentValid: false,
            contentValidationMessage: '',
        }
        this.contentInput = React.createRef();
    }

    updateName(note_name) {
        this.setState({note_name}, () => {this.validateName(note_name)});
    }

    validateName(name) {
        let errorMessage = this.state.nameValidationMessage;
        let error = this.state.nameValid;

        name = name.trim();
        if (name.length === 0) {
            errorMessage = 'Name must have at least 1 character';
            error = true;
        } else if (name.length > 20) {
            errorMessage = 'Name cannot be longer than 20 characters';
            error = true;
        } else {
            error = false;
            errorMessage = '';
        }

        this.setState({
            nameValid: !error,
            nameValidationMessage: errorMessage,
        });
    }

    validateContent(content) {
        let errorMessage = this.state.contentValidationMessage;
        let error = this.state.contentValid;

        content = content.trim();
        if (content.length === 0) {
        errorMessage = 'Content must have at least 1 character';
        error = true;
        }
        this.setState({
        contentValid: !error,
        contentValidationMessage: errorMessage
        });
    }

    handleSubmit(event) {
        event.preventDefault();

        const content = this.contentInput.current.value;
        this.validateContent(content);
        const { note_name } = this.state;

        const options = {
            method: 'POST',
            headers: {
                'content-type': 'application/json',
            },
            body: JSON.stringify({note_name, content})
        }
        fetch(`${config.API_ENDPOINT}/notes`, options)
            .then(resp => {
                if (!resp.ok) {
                    throw new Error('Something went wrong')
                }
                return resp.json();
            })
            .then(respJson => {
                this.context.addNote(respJson);
                this.props.history.push(`/note/${respJson.id}`);
            })
            .catch(error => {
                console.log(error.message);
            });
    }
    render() {
        return (
            <section>
                <header>
                    <h2>New Note</h2>
                </header>
                <form onSubmit={(event => this.handleSubmit(event))}>
                    <div>
                        <label htmlFor="note-name-input">Title</label><br />
                            <input type="text" placeholder="Note name..." id="note-name-input" name="note-name-input" value={this.state.name} onChange={event => this.updateName(event.target.value)} required />
                            {(!this.state.nameValid && this.state.nameValidationMessage) && <p className="error__message">{this.state.nameValidationMessage}</p>}<br />
                            <label htmlFor="content-input">content</label><br />
                            <textarea type="text" placeholder="content" id="note-content-input" name="note-content-input" ref={this.contentInput} required></textarea>
                            {(!this.state.contentValid && this.state.contentValidationMessage) && <p className="error__message">{this.state.contentValidationMessage}</p>}<br />
                        </div>
                    <button type="submit" disabled={!this.state.nameValid && !this.state.contentValid}>Add Note</button>
                </form>
            </section>
        )
    }
}
