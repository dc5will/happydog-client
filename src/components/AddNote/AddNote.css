.addTaskForm {
    padding: 40px;
}

.addNoteSubmit {
    margin: 40px;
    font-size: 24px;
    background-color: rgb(192, 81, 81);
    margin-bottom: 20px;
}

.addNoteSubmit:hover {
    background-color: rgb(139, 36, 36);
}

.addNoteSubmit:active {
    background-color: rgb(138, 32, 32);
    box-shadow: 0 5px #666;
    transform: translateY(2px);
}

.addNoteBannerBackground {
    background: url(../../images/pancake2.jpg) no-repeat, center;
    background-size: cover;
    opacity: 0.8;
    height: 400px;
    max-width: 100%;
}

.taskBanner {
    display: flex;
    text-shadow: 8px 8px 8px #000000;
    justify-content: center;
    margin-top: 150px;
    font-size: 54px;
    font-weight: bolder;
    font-family: 'Dancing Script';
    padding: 10px;
    opacity: 100%;
    color: white;
    text-transform: none;
}

.taskTitleInput {
    margin-top: 20px;
}

.titleLabel {
    font-weight: bold;
    font-size: 20px;
}

.AddNote {
    height: calc(100% - 130px);
}