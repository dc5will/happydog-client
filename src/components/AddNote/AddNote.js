import React, { Component } from 'react';
import NotesContext from '../../contexts/NotesContext';
import NotesApiService from '../../services/notes-api-service';



export default class AddNote extends Component {
  static contextType = NotesContext

  handleSubmit = e => {
    e.preventDefault()
    const name = e.target.name.value;
    NotesApiService.postNote(name).then((note) => this.context.addNote(note))
    this.props.history.push('/my-notes')
  }

  render() {
    return (
      <form onSubmit={(e) => this.handleSubmit(e)}>
        <label>Title:
          <br />
          <input type='text' id = 'name' ></input>
        </label>
        <br />
      <button type='submit'>Submit</button>
    </form>
    )
  }
}