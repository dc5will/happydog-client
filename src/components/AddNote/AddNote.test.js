import React from "react";
import { shallow } from "enzyme";
import AddNote from "./AddNote";
import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe("AddNote component", () => {
  it("renders without crashing", () => {
    shallow(<AddNote />);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <AddNote />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
