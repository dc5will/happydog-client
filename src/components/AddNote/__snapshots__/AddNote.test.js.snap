// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNote component renders the UI as expected 1`] = `
Array [
  <nav
    className="Header"
  >
    <h1
      className="happyDogLogo"
    >
      <a
        href="/my-notes"
        onClick={[Function]}
      >
        Happy Dog
      </a>
    </h1>
    <div
      className="Header_not-logged-in"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </nav>,
  <section
    className={
      Array [
        "Section",
        undefined,
        undefined,
      ]
    }
  >
    <form
      className="addTaskForm"
      onSubmit={[Function]}
    >
      <label>
        <header>
          <h3>
            Task Title
          </h3>
        </header>
        <br />
        <input
          className="Input taskTitleInput"
          id="name"
          type="text"
        />
      </label>
      <br />
      <button
        className="Button addNoteSubmit"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>,
  <footer
    className="footer"
  >
    <div
      className="footer_brand-container"
    >
      <span
        className="footer_author-text"
      >
        Created by William
      </span>
      <div
        className="footer_social-media"
      />
      <div
        className="copyright-text"
      >
        <p>
          Copyright Â© 2019
        </p>
      </div>
    </div>
  </footer>,
]
`;
